JSTests/stress/tdz-check-catch-read.js

```diff
@@ -0,0 +1,25 @@
+//@ runDefault
+function f(a){
+    try{
+        let b=((a=b),{t(){b}})
+    }catch{
+    }
+    a._
```

Comment:
To verify I understand the bug correctly, without this change (and with the other change), a is `JSValue()` here and not the original value?

Reply:
Yes, that's right. We evaluate `a = b` (`= JSValue()`) then `check_tdz` throws, but we already assigned